src/haddock/modules/scoring/sasascore/__init__.py

```diff
@@ -0,0 +1,153 @@
+"""
+Surface accessibility scoring module.
+
+This module performs a solvent accessibility analysis based on 
+some user-defined residues that should be buried or accessible.
+
+If a supposedly buried (resp. accessible) residue is accessible (resp. buried),
+the score should increase by one. The lower the final score the more consistent
+the model with the user data.
+
+To run this module, the user must provide a dictionary with the buried and/or accessible
+residues. The keys of the dictionary should be the chain identifiers and the values
+should be lists of residue numbers.
+
+Example:
+
+>>> resdic_buried_A: [1, 2, 3]
+>>> resdic_accessible_B: [4, 5, 6]
+"""
+from pathlib import Path
+
+from haddock.core.typing import FilePath
+from haddock.modules import get_engine
+from haddock.modules import BaseHaddockModule
+from haddock.libs.libutil import parse_ncores
+from haddock.libs.libparallel import get_index_list
+from haddock.modules.scoring.sasascore.sasascore import (
+    AccScore,
+    AccScoreJob,
+    prettify_df,
+    )
+from haddock.modules.analysis import (
+    get_analysis_exec_mode,
+    )
+from typing import Optional
+
+RECIPE_PATH = Path(__file__).resolve().parent
+DEFAULT_CONFIG = Path(RECIPE_PATH, "defaults.yaml")
+
+
+class HaddockModule(BaseHaddockModule):
+    """HADDOCK3 module to perform accessibility scoring."""
+
+    name = RECIPE_PATH.name
+
+    def __init__(self,
+                 order: int,
+                 path: Path,
+                 initial_params: FilePath = DEFAULT_CONFIG) -> None:
+        super().__init__(order, path, initial_params)
+
+    @classmethod
+    def confirm_installation(cls) -> None:
+        """Confirm module is installed."""
+        return
+
+    def _rearrange_output(self, output_name: FilePath, path: FilePath,
+                          ncores: int) -> None:
+        """Combine different tsv outputs in a single file."""
+        output_fname = Path(path, output_name)
+        self.log(f"rearranging output files into {output_fname}")
+        key = output_fname.stem.split(".")[0]
+        # Combine files
+        with open(output_fname, 'w') as out_file:
+            for core in range(ncores):
+                tmp_file = Path(path, f"{key}_" + str(core) + ".tsv")
+                with open(tmp_file) as infile:
+                    out_file.write(infile.read())
+                tmp_file.unlink()
```

Comment:
This pattern here looks very similar to what you had before in the `caprieval` module and that we changed to produce less files. See the implementation I did there: https://github.com/haddocking/haddock3/blob/main/src/haddock/modules/analysis/caprieval/capri.py#L905

You can extract the values directly from the scheduler object, no need to have these intermediate files

Reply:
Something is producing several files, and they are being merge here, right? This thing that is producing the several files is the one that should go trough the scheduler and then here instead of merging the file you just retrieve the values

Reply:
I just found out that pattern is found in many other places in the code, see here: https://github.com/haddocking/haddock3/issues/970

So please add this one to the list there and I can handle them in another PR

Reply:
I am checking this over and over and I think this induces code duplication everywhere. now caprieval has both the `rearrange_output` function (when less_io = false) and this `extract_data_from_capri_class` (when less_io = true)..both doing the same thing in a completely different manner.
Everything would be much easier if fewer output files were generated by each module.

Reply:
It does yes, but it's already part of our technological debt because of how people decided this workflow execution would be done - unfortunately nothing we can do at this point except work around it 🤷🏽

but this pattern is in many places already, just add a comment in the code and add it to this list here: https://github.com/haddocking/haddock3/issues/970

Reply:
there is something that can be done, which is simply having much fewer files produced by each analysis module. The code duplication this `less_io` pattern introduces is not worth the advantage imo. Probably something to be discussed in a code meeting. I won't add anything to that PR for the time being.

Reply:
I think I understand what you mean - here is the same case as https://github.com/haddocking/haddock3/pull/928#issuecomment-2346747672, the execution mode is being overwritten on the fly so there's no need to even consider this `less_io` parameter, the easier solution is to just set it to `local`, let the scheduler split the tasks and refactor this implementation to extract directly from the class.